sklearn/utils/estimator_checks.py

```diff
@@ -807,26 +825,30 @@ def check_fit_score_takes_y(name, Estimator):
         if func is not None:
             func(X, y)
             args = [p.name for p in signature(func).parameters.values()]
+            if args[0] == "self":
+                # if_delegate_has_method makes methods into functions
+                # with an explicit "self", so need to shift arguments
+                args = args[1:]
```

Comment:
What if some clever ass names the first argument something else than self?

It's bad style, and we would probably complain reviewing that code, so I don't think that we should cater for it.